builder error when missing type declaring generic struct array or map (ok for variable)

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
```V 0.2.4 114a341.92bfd9b```
**OS:** 
```OS: macos, macOS, 12.3.1```
```CC version: Apple clang version 13.1.6```
<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
Create an array of generic structs
```v
struct Item<T>{
	val T
}

// a := Item{42} // <int> missing error
mut arr := []Item{} // builder error:
```
**What did you expect to see?**
Generic struct declaration with missing type error
```
error: generic struct init must specify type parameter, e.g. Foo<int>
```
 
**What did you see instead?**
```
/tmp/v_501/aaastrcuterr.9191433161597575981.tmp.c:11341:63: error: use of undeclared identifier 'main__Item_T_T'
        Array_main__Item arr = __new_array_with_default(0, 0, sizeof(main__Item_T_T), 0);
                                                                     ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

With missing type in array declaration you get a build error.  The parser has a problem picking up the syntax error in the source code which results invalid C code  ```C error. This should never happen.```.  Information is for the compiler designers to have a look at. 

V should display the informative error message in both cases for the programmer to fix the source code.  
 
```v
struct Item<T>{
	val T
}

a := Item{42} 
// error: must specify type parameter, e.g. Foo<int>

// should be an error message like the one for the variable
mut arr := []Item{} 
/tmp/v_501/aaastrcuterr.4347668897322262372.tmp.c:11341:63: error: use of undeclared identifier 'main__Item_T_T'
         Array_main__Item arr = __new_array_with_default(0, 0, sizeof(main__Item_T_T), 0);
                                                                      ^
 1 warning and 1 error generated.
 ...
 ==================
 (Use `v -cg` to print the entire error message)

 builder error: 
 ==================
 C error. This should never happen.
```

--- a/vlib/v/checker/containers.v
+++ b/vlib/v/checker/containers.v
@@ -9,6 +9,22 @@ pub fn (mut c Checker) array_init(mut node ast.ArrayInit) ast.Type {
 	mut elem_type := ast.void_type
 	// []string - was set in parser
 	if node.typ != ast.void_type {
+		if node.elem_type != 0 {
+			elem_sym := c.table.sym(node.elem_type)
+			if elem_sym.kind == .struct_ {
+				elem_info := elem_sym.info as ast.Struct
+				if elem_info.generic_types.len > 0 && elem_info.concrete_types.len == 0
+					&& !node.elem_type.has_flag(.generic) {
+					if c.table.cur_concrete_types.len == 0 {
+						c.error('generic struct must specify type parameter, e.g. Foo<int>',
+							node.elem_type_pos)
+					} else {
+						c.error('generic struct must specify type parameter, e.g. Foo<T>',
+							node.elem_type_pos)
+					}
+				}
+			}
+		}
 		if node.exprs.len == 0 {
 			if node.has_cap {
 				c.check_array_init_para_type('cap', node.cap_expr, node.pos)

--- a/vlib/v/checker/tests/array_of_generic_struct_init_err.out
+++ b/vlib/v/checker/tests/array_of_generic_struct_init_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/array_of_generic_struct_init_err.vv:6:15: error: generic struct must specify type parameter, e.g. Foo<int>
+    4 |
+    5 | fn main() {
+    6 |     mut arr := []Item{}
+      |                  ~~~~
+    7 | }

--- a/vlib/v/checker/tests/array_of_generic_struct_init_err.vv
+++ b/vlib/v/checker/tests/array_of_generic_struct_init_err.vv
@@ -1,0 +1,7 @@
+struct Item<T>{
+	val T
+}
+
+fn main() {
+	mut arr := []Item{}
+}

